src/main/scala/org/clulab/habitus/utils/TsvPrinter.scala

```diff
@@ -70,27 +70,27 @@ class TsvPrinter(outputFilename: String) {
                 }
 
               if (contexts.contains(i)) {
-                for (context <- contexts(i)) {
+
                   pw.println(s"$varText\t$valText\t$norm\t$sentText\t$filename\t${
-                    context.mostFreqLoc0Sent
+                    contexts(i)(0).mostFreqLoc0Sent
                   }\t${
-                    context.mostFreqLoc1Sent
+                    contexts(i)(0).mostFreqLoc1Sent
                   }\t${
-                    context.mostFreqLoc
+                    contexts(i)(0).mostFreqLoc
                   }\t${
-                    context.mostFreqDate0Sent
+                    contexts(i)(0).mostFreqDate0Sent
                   }\t${
-                    context.mostFreqDate1Sent
+                    contexts(i)(0).mostFreqDate1Sent
                   }\t${
-                    context.mostFreqDate
+                    contexts(i)(0).mostFreqDate
                   }\t${
-                    context.mostFreqCrop0Sent
+                    contexts(i)(0).mostFreqCrop0Sent
                   }\t${
-                    context.mostFreqCrop1Sent
+                    contexts(i)(0).mostFreqCrop1Sent
                   }\t${
-                    context.mostFreqCrop
+                    contexts(i)(0).mostFreqCrop
```

Comment:
I'm concerned about this (0).  If only that one is used, why is there room for more?  Before there was a loop through potentially multiple contexts.

Reply:
Fair point. @mithunpaul08 : can you please give an example of a sentence with multiple contexts?

Reply:
'''Farmers’ sowing dates ranged from 14 to 31 July for the WS and from 3 to 11 March for the DS.'''

ODIN extracts two events for this sentence (one each for each date range) , but for the same trigger (sowing dates). That is expected right?

Anyway, coming back to the point, the contexts(0) thing is vestigial from the bug fix. Rather, what was happening was that, if there are multiple events in the same sentence, i was creating a separate context object for each mention. Thinking of it now, it is not necessary because all the context counts like LOC,CROP etc (within n sentences away) should remain same wrt to multiple events as long as they occur in the same sentence. . Mihai, your email mentions this is the correct assumption.

Anyway, the original duplication bug was caused because of a for loop (me running through the contexts) for each sorted mentions(which already had a list of multiple mentions). Let me go back create a new branch and make it just one context for all mentions in the same sentence. Functionality will remain the same. I am also in the middle of cleaning up all those long list of contexts(i).mostfreq* with more simple data structures.